Update ANSSI BP28 profiles in rhel10 product

vojtapolasek opened the pull request:
#### Description:

- revise list of excluded rules; remove some exclusions and add comments
- make some modifications to the content (I can propose them as separate PR if needed). The reason is so that the content can be tested without much noise.
    - remove dedicated ssh group owning private ssh keys because RHEL 10 no longer uses it
    - the aide.conf directive for database location has changed, I changed it in the content as well

#### Rationale:

- add content to rhel10 product

#### Review Hints:

```
./build_product rhel10
cd tests
./ds_unselect_rules.sh  ../build/ssg-rhel10-ds.xml  unselect_rules_list 
python automatus.py profile --libvirt <your_qemu_conection> <your_vm_name> --datastream /tmp/ssg-rhel10-ds.xml anssi_bp28_high
```

vojtapolasek commented:
@Mab879  @marcusburghardt  as you worked with some profile in rhel10 product recently... could you voice your opinion on rules which I made no longer excluded from the ANSSI profile? It was not clear to me why they were actually excluded. Thank you.

Mab879 commented on products/rhel10/profiles/anssi_bp28_enhanced.profile:
```diff
@@ -5,11 +5,10 @@ metadata:
         - vojtapolasek
 
-title: 'DRAFT - ANSSI-BP-028 (enhanced)'
+title: '[Draft] ANSSI-BP-028 (enhanced)'
```
Please don't use `[` it causes issues in other processes.

Mab879 commented on products/rhel10/profiles/anssi_bp28_high.profile:
```diff
@@ -5,11 +5,10 @@ metadata:
         - vojtapolasek
 
-title: 'DRAFT - ANSSI-BP-028 (high)'
+title: '[Draft] ANSSI-BP-028 (high)'
```
Please don't use [ it causes issues in other processes.

Mab879 commented on products/rhel10/profiles/anssi_bp28_intermediary.profile:
```diff
@@ -5,11 +5,10 @@ metadata:
         - vojtapolasek
 
-title: 'DRAFT - ANSSI-BP-028 (intermediary)'
+title: '[Draft] ANSSI-BP-028 (intermediary)'
```
Please don't use [ it causes issues in other processes.

Mab879 commented on products/rhel10/profiles/anssi_bp28_minimal.profile:
```diff
@@ -5,11 +5,10 @@ metadata:
         - vojtapolasek
 
-title: 'DRAFT - ANSSI-BP-028 (minimal)'
+title: '[Draft] ANSSI-BP-028 (minimal)'
```
Please don't use [ it causes issues in other processes.

Mab879 commented on products/rhel10/profiles/anssi_bp28_high.profile:
```diff
@@ -5,11 +5,10 @@ metadata:
 description: |-
-    This is a draft profile for experimental purposes.
-    This draft profile contains configurations that align to ANSSI-BP-028 v2.0 at the high hardening level.
+    This profile contains configurations that align to ANSSI-BP-028 v2.0 at the high hardening level.
```
Any reason for removing the draft wording that was already there?

vojtapolasek commented:
Hello @Mab879  and thank you. Issues you highlighted were caused by copying over files from RHEL 9. I fixed that.

marcusburghardt commented on products/rhel10/profiles/anssi_bp28_enhanced.profile:
```diff
@@ -22,32 +22,33 @@ description: |-
     - anssi:all:enhanced
-    # Following rules are incompatible with the rhel10 product
-    - '!partition_for_opt'
+    - var_password_hashing_algorithm=SHA512
```
Please, update to `yescrypt`.

marcusburghardt replied:
```suggestion
    - var_password_hashing_algorithm=yescrypt
```

marcusburghardt commented on products/rhel10/profiles/anssi_bp28_enhanced.profile:
```diff
@@ -22,32 +22,33 @@ description: |-
-    # Following rules are incompatible with the rhel10 product
-    - '!partition_for_opt'
+    - var_password_hashing_algorithm=SHA512
+    - var_password_pam_unix_rounds=65536
```
```suggestion
    - var_password_pam_unix_rounds=11
```
In alignment to `yescrypt`

vojtapolasek replied:
I implemented this change in form of 758c5e3f1f725101925e9c7ccb6e498cc1ba0ac7

marcusburghardt reviewed (changes requested):
RHEL10 is using `yescrypt` by default instead of SHA512. Could you update the variables, please?

marcusburghardt commented:
> @Mab879 @marcusburghardt as you worked with some profile in rhel10 product recently... could you voice your opinion on rules which I made no longer excluded from the ANSSI profile? It was not clear to me why they were actually excluded. Thank you.

It was also not clear to me the reason some the rules were excluded. But your updates make sense to me.

marcusburghardt commented:
> > @Mab879 @marcusburghardt as you worked with some profile in rhel10 product recently... could you voice your opinion on rules which I made no longer excluded from the ANSSI profile? It was not clear to me why they were actually excluded. Thank you.
> 
> It was also not clear to me the reason some the rules were excluded. But your updates make sense to me. You only need to ensure these rules have CCEs for RHEL 10.

@vojtapolasek the CCEs errors are still present.

vojtapolasek commented:
@marcusburghardt  I added missing CCEs.Hello @Mab879  and thank you. Issues you highlighted were caused by copying over files from RHEL 9. I fixed that.

marcusburghardt reviewed (changes requested):
My local tests worked fine testing the profile and remediating with Ansible. However, when remediating with Bash the automatus test finishes with error. I isolated the problem and it is related to the `service_sssd_enabled` rule. I saw some issues with this rule and intend to fix them soon. But for now, could you also remove this rule from the RHEL 10 Draft profiles, please?

marcusburghardt commented:
> service_sssd_enabled

@vojtapolasek , I would recommend to remove the following rules from RHEL 10 profiles for now so we can merge this PR:

- sssd_enable_pam_services
- sssd_ldap_configure_tls_reqcert
- sssd_ldap_start_tls

I am working on these rules in https://github.com/ComplianceAsCode/content/pull/12378 but it will still take some time to organize all these SSSD related rules.

marcusburghardt reviewed (approved):
Nice work @vojtapolasek . Thanks
